src/core/tests/Test_StatusHandler.py

```diff
@@ -358,6 +358,34 @@ def test_set_current_operation_auto_assess_non_assessment(self):
         self.assertRaises(Exception,
                           lambda: self.runtime.status_handler.set_current_operation(Constants.INSTALLATION))
 
+    def test_sort_packages_by_classification_and_state(self):
+        with self.runtime.env_layer.file_system.open("../../extension/tests/helpers/AssessmentSummary.json", 'r') as file_handle:
+            assessment_patches = json.load(file_handle)["patches"]
+            assessment_patches_sorted = self.runtime.status_handler.sort_packages_by_classification_and_state(assessment_patches)
+            self.assertEqual(assessment_patches_sorted[0]["name"], "test-package-3")
+            self.assertEqual(assessment_patches_sorted[1]["name"], "test-package-4")
+            self.assertEqual(assessment_patches_sorted[2]["name"], "test-package-7")
+            self.assertEqual(assessment_patches_sorted[3]["name"], "test-package-1")
+            self.assertEqual(assessment_patches_sorted[4]["name"], "test-package-5")
+            self.assertEqual(assessment_patches_sorted[5]["name"], "test-package-2")
+            self.assertEqual(assessment_patches_sorted[6]["name"], "test-package-6")
+
+        with self.runtime.env_layer.file_system.open("../../extension/tests/helpers/InstallationSummary.json", 'r') as file_handle:
+            installation_patches = json.load(file_handle)["patches"]
+            installation_patches_sorted = self.runtime.status_handler.sort_packages_by_classification_and_state(installation_patches)
+            self.assertEqual(installation_patches_sorted[0]["name"], "test-package-6")
+            self.assertEqual(installation_patches_sorted[1]["name"], "test-package-12")
+            self.assertEqual(installation_patches_sorted[2]["name"], "test-package-11")
+            self.assertEqual(installation_patches_sorted[3]["name"], "test-package-10")
+            self.assertEqual(installation_patches_sorted[4]["name"], "test-package-9")
+            self.assertEqual(installation_patches_sorted[5]["name"], "test-package-8")
+            self.assertEqual(installation_patches_sorted[6]["name"], "test-package-7")
+            self.assertEqual(installation_patches_sorted[7]["name"], "test-package-5")
+            self.assertEqual(installation_patches_sorted[8]["name"], "test-package-4")
+            self.assertEqual(installation_patches_sorted[9]["name"], "test-package-3")
+            self.assertEqual(installation_patches_sorted[10]["name"], "test-package-2")
+            self.assertEqual(installation_patches_sorted[11]["name"], "test-package-1")
```

Comment:
It's a little bit of work now but for long-term ease of reading, please append the classification and state as a comment at the end of each line to make clear what the outcome was. Ideally tabbed correctly so it reads like a table, and helps for fast comparison against the input json.

Reply:
Good idea, added!